Fixes #19858

fortuin opened the pull request:
This PR removes spaces from the `equation` argument to the `tf.einsum` function, such that the function can handle spaces in the same way as `np.einsum` does.

fortuin committed:
Fixes #19858

fortuin commented:
I signed it!

drpngx commented:
Could you add a test for that?

fortuin commented:
Sure, I can do that. Shall I use `tf.test` for it? Where do I add the test?

drpngx commented:
I think you just need to add something here:
https://www.github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/special_math_ops_test.py#L155

fortuin committed:
Add tests for PR 19859

Add tests for handling of spaces in the equation parameter of the tf.einsum() function

fortuin commented:
Ok, I did that.

drpngx commented:
Thanks!

fortuin commented:
I see, the testing framework assumes that there are no spaces while figuring out the shapes of the tensors that are randomly generated as inputs for the tests. Let me try to fix this...

fortuin committed:
Fix space handling in random tensor generation

Relating to PR #19859, the tf.einsum() function is now supposed to also handle spaces in the equation parameter. Hence, the test relating to that should also just consider the alphabetical parts of the equation to figure out which shape the random input tensor for the test should have.

fortuin commented:
Can you run the tests again?

drpngx commented:
Oh, this was sent against `1.8`. You have to send it against master. Could you do that in another PR and CC me on it? Thanks.

fortuin commented:
Oops, sorry for that. I created a PR into master in #19980.

tensorflowbutler commented:
Nagging Assignee @drpngx: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

fortuin commented:
This was settled in PR #19980.
